### PR TITLE
adds Pursuit search to the completion

### PR DESCRIPTION
**Pull request by kRITZCREEK**
changes the completion API from `complete stub` to `complete stub (File|Project|Pursuit)` where File and Project do the same for now and Pursuit fetches completion results from Pursuit

### CODE REVIEW

**Comment by kRITZCREEK** [comment]
@epost Some reviewing would be appreciated. Since everything is synchronous this does block for the entire HTTP request. Maybe we should consider asynchronous options for our new protocol?

File: lib/Purescript/Ide/Command.hs [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
 import qualified Data.Text        as T
 import           Text.Parsec
 import           Text.Parsec.Text
 
+data Level = File | Project | Pursuit deriving(Show, Eq)
+
 data Command =
   TypeLookup Text
-  | Completion Text
+  | Completion Text Level
```
Comment (epost) [inline lib/Purescript/Ide/Command.hs]: Maybe `s/Completion/Complete/`? or keep that for another commit?
